### PR TITLE
Send woocommerce_onboarding_profile.completed when submitting the business info step

### PR DESCRIPTION
**Pull request by lsl**
#### Changes proposed in this Pull Request

* Send the `completed: true` value when submitting the business info step.

#### Testing instructions

Final testing requires deactivating and uninstalling woo but you can skip this for most testing

```
wp plugin deactivate woocommerce
wp plugin uninstall woocommerce
```

Final testing should also include testing against a reverted / fresh site to ensure complete is copied over during transfer too (I think this is working so far but haven't confirmed)

Between tests of the installer flow it's a good idea to delete the option via cli

```
wp option delete woocommerce_onboarding_profile
```
Start the test from either the landing page or just go to the first step directly:

- http://calypso.localhost:3000/start/woocommerce-install/?site=wooptestupgrade.wpcomstaging.com
- Enter your address
- When confirming you should see the POST against `/sites/197963111/settings` with changed address values.
- You'll now be on the business info step. 
- When you submit this step, a `completed: true` value should be included in the /sites/settings POST request regardless of whether you changed anything on the options presented or just clicked submit early. At the moment it seems like both scenarios fail to include the completed: true value.
- After submitting you'll be taken to the transfer step where woo will be "installed" or at least checked in our test case
- After this check completes you're taken to the woocommerce admin screen, you should be left on wc admin without being redirected to the setup wizard.

This video shows sort of what I'm testing / how a bit clearer:

https://user-images.githubusercontent.com/811776/150071960-a56cf4ee-dfb4-43b1-ba3b-ce2d493e2287.mp4

The grey screen wsod at the end there is due to a JS error because we send product types as a string instead of an array, this will be fixed separately in https://github.com/Automattic/wp-calypso/issues/60171. If you need to work around it (and get it stuck in local storage state like I did, you can clear your local state out and it shouldn't be a problem as long as you delete it from site options too.

Related to #60228

### CODE REVIEW

File: client/signup/steps/woocommerce-install/hooks/use-site-settings/index.tsx [hunk 1]
```diff
@@ -81,22 +81,25 @@ export function useSiteSettings( siteId: number ) {
...
-			( acc, settingKey ) => ( { ...acc, [ settingKey ]: settings[ settingKey ] } ),
-			{}
-		);
-
-		dispatch( saveSiteSettings( siteId, newSettings ) );
-		setEditedSettings( [] );
-	}, [ dispatch, editedSettings, settings, siteId ] );
-
-	return { save, update, get };
+		setEditedSettings( ( state ) => {
```
Comment (lsl) [inline client/signup/steps/woocommerce-install/hooks/use-site-settings/index.tsx]: This doesn't work just wanted to get something up to explain the problem better, I thought this was working for a bit but it was due to a completed:true value leaking into existing state from a previous fix attempt that only worked in some situations.

File: client/signup/steps/woocommerce-install/step-business-info/index.tsx [hunk 2]
```diff
@@ -176,9 +176,9 @@ export default function StepBusinessInfo( props: WooCommerceInstallProps ): Reac
 						<SupportCard />
 						<StyledNextButton
 							onClick={ () => {
-								dispatch( submitSignupStep( { stepName: 'business-info' } ) );
 								updateOnboardingProfile( 'completed', true );
 								save();
+								dispatch( submitSignupStep( { stepName: 'business-info' } ) );
```
Comment (lsl) [inline client/signup/steps/woocommerce-install/step-business-info/index.tsx]: This doesn't have any bearing on the situation but moving it down makes some sense.

File: client/signup/steps/woocommerce-install/step-business-info/index.tsx [hunk 3]
```diff
@@ -176,9 +176,9 @@ export default function StepBusinessInfo( props: WooCommerceInstallProps ): Reac
 						<SupportCard />
 						<StyledNextButton
 							onClick={ () => {
-								dispatch( submitSignupStep( { stepName: 'business-info' } ) );
 								updateOnboardingProfile( 'completed', true );
 								save();
```
Comment (lsl) [inline client/signup/steps/woocommerce-install/step-business-info/index.tsx]: This `updateOnboardingProfile('completed', true);` doesn't change `editedSettings` in time for the `save()` function to see it up in ln84.

**Review by DustyReagan — APPROVED** [review]
This is working great!

- Tested blank profile form. ✅ 
- Tested full profile form. ✅ 
- Tested going back and editing my email then proceeding with blank profile. ✅ 
- Tested going back and editing my email then proceeding with new profile data. ✅ 

Having `completed : true` is skipping the wc-admin setup wizard. ✅

:shipit:

**Pull request merged**